### DISCUSSION TITLE
pip installation error

### DISCUSSION THREAD

**Discussion by AragornHorse** [pypr/pysph#362] [Q&A] [answered]
I have the proper version of visual c++14.0 and python3.9
and try to install with '$ pip install https://github.com/pypr/pysph/zipball/master' or '$ pip install PySPH'
but the result is 'Cython.Compiler.Errors.CompileError: pysph/base/box_sort_nnps.pyx'
it seems that something wrong with the octree.cpp ,like

pysph/base/octree.cpp(16176): error C3861: '_Pragma': identifier not found
pysph/base/octree.cpp(16176): error C2059: syntax error: ';'

how to solve

**Comment by mat4133** [comment, marked as answer]
I had this issue and switching to MSVC 2019 fixed the problem

**Reply by AragornHorse** [reply]
solved , thanks
